### ISSUE TITLE
yaml generator does not put daemonset in namespace kube-system

### ISSUE THREAD

**Issue by deitch** [weaveworks/weave#3006]
## What you expected to happen?

yaml generator would generate yml that puts daemonset and other components in namespace `kibe-system`. Existing static config from git.io has it.

From static git.io:

```yml
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: weave-net
  namespace: kube-system
```

From yaml generator:

```yml
  - apiVersion: extensions/v1beta1
    kind: DaemonSet
    metadata:
      name: weave-net
      annotations:
        cloud.weave.works/launcher-info: |-
          {
            "server-version": "master-b9cda10",
            "original-request": {
              "url": "/k8s/v1.6/net.yaml?env.IPALLOC_RANGE=10.79.0.0/16",
              "date": "Mon Jun 05 2017 11:17:52 GMT+0000 (UTC)"
            },
            "email-address": "support@weave.works"
          }
      labels:
        name: weave-net
```

## What happened?

no namespace applied to daemonset, so it just goes to default.

## How to reproduce it?

* `curl -L https://git.io/weave-kube-1.6` - the output will contain the namespace
* `curl "https://cloud.weave.works/k8s/v1.6/net.yaml?env.IPALLOC_RANGE=10.79.0.0/16&env.EXPECT_NPC=0"` - the output will have no namespace

## Anything else we need to know?
Nope.

## Versions:

(not really relevant here, since it is the output of yaml generator, but happy to oblige)
```
$ weave version
1.9.7
$ docker version
1.12.3
$ uname -a
Linux ip-10-78-21-180.ec2.internal 4.7.3-coreos-r2 #1 SMP Sun Jan 8 00:32:25 UTC 2017 x86_64 Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz GenuineIntel GNU/Linux
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by marccarre** [comment]
Hi @deitch, the lack of namespace is expected when using the Launch Generator and documented in `master` (new version of the doc will be released with 2.0): https://github.com/weaveworks/weave/blob/master/site/kube-addon.md.
After internal discussions, the reasoning was that namespaces should be made explicit and not hidden inside the YAML.

**Comment by deitch** [comment]
Thanks @marccarre . So basically:

1. namespaces are not defined in yaml generator intentionally
2. if you want it all in one namespace, do it via kubectl (like I did `kubectl -n kube-system -f ...`)
3. Docs will state that.

Should it be an option for the yaml generator? `&namespace=kube-system`?

**Comment by marccarre** [comment]
1. Correct. And this is because we wanted to make the namespace more explicit, and also harmonise the way the Launch Generator works (it generates YAML for other products like Flux, Scope, Cortex, which do not necessarily need `kube-system` as a namespace).
2. Correct. At least for now.
3. Correct.

> Should it be an option for the yaml generator? &namespace=kube-system?

Maybe, maybe not, we're still refining this, but we've indeed considered such a query parameter.
There is also this [annotation](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) we may need for Net, but that requires `kube-system`, so we may actually revert to hardcoding the `kube-system` namespace in the generated YAML for Net and undo 1.

In summary: this is very much work in progress, which is why we haven't advertised the Launch Generator just yet for Net.

**Comment by deitch** [comment]
OK, thanks.

**Comment by bboreham** [comment]
This change is still confusing people; re-opening to track progress.
